refactor: 프롬프트 빌더 리펙토링 (#145)

Hoyoung9509 opened the pull request:
<!-- 
아래 중 하나의 타입을 선택해 제목 앞에 붙여주세요.
feat: 새로운 기능 추가
fix: 버그 수정
refactor: 코드 리팩토링
docs: 문서 추가 및 수정
style: 포맷팅, 세미콜론 등 비즈니스 로직 영향 없는 변경
test: 테스트 코드 관련 작업
chore: 빌드/패키지 매니저 설정 등 기타 변경
-->

# 제목: 프롬프트 빌더 리펙토링

---

## ✅ PR 종류
- [ ] feat (새로운 기능 추가)
- [ ] fix (버그 수정)
- [ ] refactor (코드 리팩토링)
- [ ] docs (문서 추가 및 수정)
- [ ] style (비즈니스 로직 영향 없는 변경)
- [ ] test (테스트 코드 관련 작업)
- [ ] chore (빌드/설정 등 기타 변경)

---

## 📝 작업 내용
<!-- 이번 PR에서 작업한 내용을 간단히 정리 -->
프롬프트 빌더 리펙토링
---

## 🔍 변경 이유
<!-- 왜 이 작업을 했는지, 해결하려는 문제는 무엇인지 설명 -->
기존 사용자의 요청이 그대로 답변에 옮겨지던 ISSUE 조치
---

## 📸 스크린샷 (선택)
<!-- UI 변경 시 스크린샷 첨부 -->

---

## ✅ 체크리스트
- [ ] 코드 동작 확인
- [ ] 관련 테스트 작성/수정
- [ ] 문서 업데이트
- [ ] 리뷰 반영

choichanwoo001 committed:
refactor: 라우터 파일 정리 및 중복 기능 제거 (#133)

* refactor: 필요없는 라우트들 삭제

* refactor(alimtalk): 하드코딩된 검증 로직을 AI 기반으로 변경

변경사항:
- generate_alternatives() 함수 삭제 (70여 줄의 하드코딩된 대안 추천)
- generate_suggestion_for_error() 함수 삭제 (하드코딩된 수정 제안)
- extract_variable_name_from_error() 함수 삭제 (변수 추출 로직)
- convert_stage_to_korean() 함수 삭제 (한국어 변환 로직)

개선사항:
- alternatives 필드를 빈 객체로 설정하여 AI에서 생성하도록 변경
- suggestion 필드를 일반적인 메시지로 변경
- variable_name을 None으로 고정
- stage 필드에서 원본 값 직접 사용
- rejected_variables를 빈 리스트로 설정

* refactor: AI-백엔드 간 데이터 구조 통일

- ValidationResponse 모델 구조 변경 (백엔드 호환)
  * success, message, rejected_variables, validation_errors, alternatives 필드
- alimtalk_service.py: ValidationResult → ValidationResponse 변환 로직 추가
- alimtalk_routes.py: convert_to_backend_format 함수 제거
- 변환 함수 없이 직접 ValidationResponse 반환하도록 수정

* doc: 필요한 주석 추가

* refactor: 모델 파일 구조 개선 - AI 모델들을 alimtalk_models.py로 통합

- ai_models.py 파일 제거하여 모델 관리 단순화
- ChatRequest, ChatResponse, TemplateModificationRequest, TemplateModificationResponse를 alimtalk_models.py로 이동
- ai_routes.py에서 import 경로를 alimtalk_models로 변경
- 모든 모델을 한 곳에서 관리하여 프로젝트 구조 개선

* refactor: 필요없는 디렉토리, 초기화파일 삭제

* fix: 템플릿 수정하기위해 필요했던 router 주석 해제 및 임포트 수정

* refactor: 사용하지않는 메서드들 다 삭제

* refactor: 필요없는 파일 삭제

* refactor: ai_routes.py에서 의존성 주입 패턴 적용

- 직접 OpenAIService 초기화 코드 제거
- dependencies.py의 get_openai_service 함수 사용
- /ai/openai/chat, /ai/template/modify 엔드포인트에 Depends() 적용
- ai_routes.py와 template_routes.py 간 서비스 관리 방식 통일
- 메모리 효율성 및 일관성 개선

* refactor: OpenAIService 정리 및 단일 책임 원칙 적용

- OpenAIService에서 사용하지 않는 메서드들 제거 (text_completion, embeddings, batch_embeddings)
- chat_completion 메서드만 남겨 범용 OpenAI 클라이언트로 단순화
- AlimtalkValidationService에서 범용 기능 제거하여 검증 서비스 역할에 집중
- 단일 책임 원칙(SRP) 준수로 설계 개선
- 모든 파일의 import 구문 정리

* refactor: 사용하지않는 프롬프트 파일 삭제

* refactor(semantic_validator): 코드 품질 개선 및 로직 수정

주요 변경사항:
- 하이퍼파라미터 상수화: REJECT_THRESHOLD(0.6), REVIEW_THRESHOLD(0.4) 등
- 실제 검색 결과 기반 점수 계산으로 변경 (기존 하드코딩 제거)
- __init__ 간소화: OpenAI 의존성 제거, 불필요한 로깅 삭제
- 불필요한 메서드 제거: _debug_collections(), _search() 래퍼
- 컬렉션 이름 수정: blacklist/denied_templates → approved_templates/pulblic_templates

* fix: 패키지 추가

---------

Co-authored-by: MinYeong Park <116958681+judy78799@users.noreply.github.com>

Commit:
chore: 채팅 결과 창 레이아웃 수정

Commit:
chore: 카테고리 조회 방식 변경 리스트 -> DB 조회/생성 처리

Hoyoung9509 committed:
refactor: 템플릿 변경 (#138)

* feat: 프론트 수정 시도 Beta v1.0

* feat: 프론트 템플릿 수정 Open Beta v0.01

* feat: 프론트 템플릿 수정 Open Beta v0.02

* feat: 프론트 템플릿 수정 Open Beta v0.03

Hoyoung9509 committed:
Delete 클론서비스 예시2.jpg

daamont committed:
Feature: 템플릿 수정 후 저장 기능 (#135)

* refactor: 채팅 수정 부분 프롬프트 및 출력 개선

* feat: 수정 후 제출 시 DB 저장
- 저장 DTO 추가
- 저장하는 코드 추가
- 저장 후 생성된 templateId를 검증으로 전달

* fix: 수정 시 500 오류 수정
 Changes to be committed:
	modified:   ai/main.py
	modified:   ai/routers/ai_routes.py

* feat: category 테이블에 칼럼 추가 및 seed 데이터 삽입
- category 테이블에 code, is_active 컬럼 추가
- 기본 카테고리 데이터 삽입

* refactor: 템플릿 수정 후 저장 기능 개선 및 카테고리 저장 기능
- 템플릿 저장 프로세스 개선: 수정 후 저장을 단계별 버튼 라벨 및 상태 관리
- DB에 존재하지 않은 카테고리가 있다면 추가하는 기능 구현

* feat: 카테고리 칼럼 추가 및 사용된 카테고리 카운트
- usage_count : 현재 카테고리를 얼마나 사용했는지 파악하기 위함
- created_by : AI 가 생성한 카테고리인지, 운영자(매니저)가 직접 추가한 카테고리인지 구분하기 위함
- created_at : 카테고리 생성 시점을 파악하기 위함

* fix: 변수가 없을 때 변수 저장이 안되도록 수정
 Changes to be committed:
	modified:   back/src/main/java/com/example/service/TemplateService.java

* chore: 충돌 해결했을 때 생긴 필요 없는 코드 삭제

* refactor: 코드리뷰 반영 및 수정 프롬프트 작업지침 추가
- front에서 dict로 있던걸 list로 수정
- 변수명 #{변수명}으로 수정
- 대화 맥락 파악을 위한 프롬프트 및 로그 추가

* fix: variable이 key-value 값인 걸 list로 수정 및 수정 횟수 변경
- 수정횟수 리셋 테스트 함수 값을 10으로 수정
- 변수가 key-value 값인걸 list로 변경

Hoyoung9509 committed:
feat: https(SSL 인증) 기능 추가 (#140)

Hoyoung9509 committed:
feat: https SSL 인증 추가 (#141)

* feat: https(SSL 인증) 기능 추가

* feat: https(SSL 인증) 기능 추가

Hoyoung9509 committed:
Update deploy.sh

Hoyoung9509 committed:
Merge develop branch into template_pt branch

Hoyoung9509 committed:
refactor: 생성 프롬프트 빌더 리펙터링. 카카오 가이드 준수, 생성 템플릿 로직 강화.

Hoyoung9509 committed:
refactor: 생성 프롬프트 빌더 리펙터링. 카카오 가이드 준수, 생성 템플릿 로직 강화.

Hoyoung9509 committed:
refactor: 생성 프롬프트 빌더 리펙터링

judy78799 commented on ai/templateEngine/prompts/builders.py:
```diff
@@ -20,502 +20,572 @@ def _build_hint_messages(self) -> List[Dict]:
+[주의사항이나 추가 설명]
+
+[마무리 인사] - 긍정적 종료
+감사합니다. / 궁금한 점이 있으시면 언제든 문의해 주세요.
```
감사합니다. 이 부분을 뒤로 빼셔야 될 것 같습니다.

judy78799 commented on ai/templateEngine/prompts/builders.py:
```diff
@@ -20,502 +20,572 @@ def _build_hint_messages(self) -> List[Dict]:
+3. **완전한 정보**: 관련된 모든 필수 정보 포함 (시간, 장소, 연락처 등)
+4. **실용적 구조**: 받는 사람이 바로 이해할 수 있는 명확한 구조
+5. **비즈니스 요소**: 문의처, 수신거부, 변경/취소 안내 등 실제 필요한 요소 포함
+
```
실용적인 부분에만 초점이 잡혀있어서 톤 앤매너 부분만 추가하시면 될 것 같아요

judy78799 commented on ai/templateEngine/nodes.py:
```diff
@@ -354,7 +376,8 @@ async def generate_with_reference_node(state: TemplateGenerationState) -> Dict[s
         messages = prompt_builder.build()
-        template = await state["openai_service"].chat_completion(messages)
+        raw_template = await state["openai_service"].chat_completion(messages)
+        template = clean_template_response(raw_template)
```
``` 부분은 시스템 프롬프트에 작성되어 있다면 해당 메서드는 사용안해도 될 듯 합니다

choichanwoo001 committed:
refactor: 검증 최종 수정 (#146)

* feat: 백엔드 모델을 문제 영역 기반으로 재설계

- ProblemArea 모델 추가: 문제 영역의 구체적인 정보 저장
- ValidationResponse를 문제 영역 기반으로 변경
- 변수 중심에서 문제 영역 중심으로 전환

* feat: 검증 서비스에서 문제 영역 추출 로직 구현

- 오류를 문제 영역으로 자동 변환
- 구체적인 문제 텍스트와 위치 추출
- AI 기반 대안 생성 유지
- final_result도 validation_results에 포함하도록 수정

* fix: 백엔드 라우터 오류 응답을 문제 영역 기반으로 수정

- 기존 rejected_variables, validation_errors 구조 제거
- problem_areas, validation_stage, total_errors, total_warnings로 변경

* feat: 프론트엔드 사이드바를 문제 영역 기반으로 수정

- 변수 중심에서 문제 영역 중심으로 전환
- 문제 영역 목록 표시 및 클릭 이벤트 구현
- 영역별 분류: 특정 문구, 문단, 전체 템플릿
- 직관적인 문제 영역 표시 및 대안 선택 UI

* fix: 변수 검증 로직 개선

- 검증 파이프라인에서 detected_variables 자동 추출 추가
- 변수 검증 프롬프트에 명확한 지침 추가
- 변수가 있음에도 '변수가 사용되지 않음' 오류 해결

* feat: AI 템플릿 생성에서 변수 형태를 #{변수명}으로 통일

- 모든 AI 생성 프롬프트에서 #{변수명} 형태 강제
- {변수명} 형태 사용 금지 명시
- 변수 추출 함수에서 #{변수명} 패턴만 인식하도록 수정

* fix: 창의성 낮췄음

* refactor: 프롬프트 폴더 안에 파일로 분리

* fix: 사용되지않고있는 pydantic model 삭제, import 정리

* feat: 알림톡 검증 시스템 개선 및 AI 대안 생성 기능 추가

- main.py: 알림톡 검증 서비스 초기화 로직 추가 및 로깅 개선
- alimtalk_routes.py: 검증 요청 처리 로직 개선 및 오류 응답 구조화
- alimtalk_service.py: 문제 영역 생성 로직 개선 및 AI 대안 생성 기능 추가
- alternative_generation_prompt.py: 검증 규칙 기반 대안 생성 프롬프트 개선

* feat: 백엔드 DTO 구조 개선 및 검증 요청 필드 추가

- TemplateRequestDto: 사용자 메시지 필드 추가 및 검증 어노테이션 적용
- TemplateResponseDto: 엔티티 변환 로직 개선 및 관계 데이터 매핑 추가
- TemplateValidationRequestDto: 변수 정보 필수 필드로 변경 및 검증 강화
- Var 엔티티: 변수 키 필드 검증 및 빌더 패턴 적용

* feat: AI 서비스 통신 로직 개선 및 템플릿 검증 처리 강화

- AIService: FastAPI 서버 통신 로직 개선 및 오류 처리 강화
- TemplateService: 검증 결과 처리 로직 개선 및 문제 영역 추출 기능 추가
- 검증 단계별 오류 처리 및 대안 제공 로직 구현

* feat: 프론트엔드 UI 개선 및 검증 결과 표시 기능 강화

- index.ts: API 타임아웃 설정 조정 및 검증 요청 구조 개선
- KakaoPreviewComponent: 문제 영역 하이라이트 및 변수 표시 기능 개선
- RejectionSidebarComponent: AI 생성 대안 표시 및 선택 기능 구현
- TemplateResultView: 검증 결과 처리 로직 개선 및 사용자 경험 향상

* feat: AI 서비스 통신 로직 개선 및 템플릿 엔진 최적화

- alimtalk_service.py: 대규모 리팩토링 (328줄 추가, 207줄 삭제)
  - 알림톡 서비스 로직 개선 및 에러 처리 강화
  - API 통신 최적화 및 응답 처리 로직 개선
- chromadb_service.py: 벡터 데이터베이스 서비스 개선
  - 검색 성능 최적화 및 쿼리 로직 개선
- openai_service.py: OpenAI API 서비스 개선
  - API 호출 최적화 및 에러 핸들링 강화
- templateEngine/nodes.py: 템플릿 노드 처리 로직 개선
- templateEngine/prompts/: 프롬프트 시스템 대폭 간소화
  - message_analyzer_prompts.py: 461줄 삭제로 대폭 간소화
  - builders.py: 프롬프트 빌더 로직 최적화
  - __init__.py: 모듈 구조 개선

* feat: 검증 시스템 개선 및 프롬프트 최적화

- constraint_validator.py: 제약 조건 검증 로직 대폭 개선 (69줄 추가, 58줄 삭제)
  - 검증 알고리즘 최적화 및 성능 향상
  - 에러 처리 및 로깅 시스템 개선
- validator_pipeline.py: 검증 파이프라인 로직 개선
- 프롬프트 시스템 전면 개선:
  - alternative_generation_prompt.py: 대안 생성 프롬프트 최적화
  - informational_message_prompt.py: 정보성 메시지 프롬프트 개선
  - template_writing_prompt.py: 템플릿 작성 프롬프트 강화
  - variable_usage_prompt.py: 변수 사용 검증 프롬프트 개선
  - __init__.py: 프롬프트 모듈 구조 정리

* refactor: 사용하지 않는 standardized_template_prompt.py 파일 삭제

- standardized_template_prompt.py: 33줄 삭제
  - 더 이상 사용되지 않는 표준화된 템플릿 프롬프트 제거
  - 코드베이스 정리 및 유지보수성 향상

* feat: 백엔드 DTO 구조 개선 및 템플릿 서비스 로직 강화

- TemplateValidationResponseDto.java: 검증 응답 DTO 구조 개선
  - 새로운 필드 추가 및 응답 구조 최적화
- Var.java: 변수 엔티티 구조 개선
  - 불필요한 필드 제거 및 엔티티 최적화
- TemplateService.java: 템플릿 서비스 로직 대폭 강화 (24줄 추가)
  - 새로운 비즈니스 로직 추가
  - 검증 처리 로직 개선
  - API 응답 처리 최적화

* feat: 프론트엔드 UI 개선 및 검증 결과 표시 기능 강화

- TemplateResultView.vue: 대규모 UI 개선 (652줄 추가, 75줄 삭제)
  - 검증 결과 표시 UI 전면 개선
  - 사용자 경험 향상을 위한 인터랙션 강화
  - 반응형 디자인 및 레이아웃 최적화
- KakaoPreviewComponent.vue: 카카오 알림톡 미리보기 컴포넌트 개선
  - 미리보기 기능 강화 및 UI 개선
- RejectionSidebarComponent.vue: 거부 사유 사이드바 컴포넌트 개선
  - 사용자 피드백 표시 기능 강화
- api/index.ts: API 통신 로직 개선
  - 백엔드와의 통신 최적화

* feat: 템플릿 정리 서비스 및 문제 추출 프롬프트 추가

- template_cleanup_service.py: 새로운 템플릿 정리 서비스 추가
  - 템플릿 데이터 정리 및 최적화 기능
  - 불필요한 데이터 제거 및 구조화 개선
- problem_extraction_prompt.py: 새로운 문제 추출 프롬프트 추가
  - 템플릿에서 문제점을 자동으로 추출하는 AI 프롬프트
  - 검증 과정에서 문제점 식별 기능 강화

* refactor: 템플릿 생성 시 불필요한 정리 서비스 제거

- generate_with_reference_node: 템플릿 정리 서비스 사용 부분 삭제
- search_public_and_generate_node: 템플릿 정리 서비스 사용 부분 삭제
- 생성 프롬프트에서 이미 깔끔한 템플릿을 생성하므로 별도 정리 불필요
- 템플릿 정리 서비스는 대안 생성/수정 과정에서만 사용하도록 최적화

* refactor: 검증 시스템 프롬프트 중복 제거 및 재사용성 향상

- system_prompts.py: 공통 시스템 프롬프트 파일 신규 생성
  - INFORMATIONAL_MESSAGE_SYSTEM_PROMPT: 정보성 메시지 검증용
  - VARIABLE_USAGE_SYSTEM_PROMPT: 변수 사용 규칙 검증용
  - TEMPLATE_WRITING_SYSTEM_PROMPT: 템플릿 작성 규칙 검증용
  - get_system_prompt() 함수로 프롬프트 타입별 반환 기능
- constraint_validator.py: 중복된 시스템 프롬프트 제거
  - 3개 chat_completion 호출에서 중복된 긴 시스템 프롬프트 제거
  - get_system_prompt() 함수 호출로 간소화
  - 코드 중복 제거 및 유지보수성 향상

* chore: 변경된 목록에서 제거하기위한 커밋

* refactor: ChromaDB 서비스 템플릿 검색 함수 통합 및 코드 정리

- search_public_templates, search_approved_templates 함수를 search_templates 공통 함수로 통합
- 컬렉션 이름을 인자로 받아 동적 검색 지원
- List[Dict] 반환 타입으로 통일 (max_similarity는 필요시 결과에서 계산)
- 중복 import 제거 및 코드 정리
- nodes.py, semantic_validator.py의 모든 호출부를 새로운 공통 함수로 변경
- 기존 호환성 유지하면서 코드 중복 제거로 유지보수성 향상

* fix: 이전 커밋 내용인데 반영 안됨;;

* fix: TemplateService.java 문법 오류 수정

- 주석 처리된 코드 블록에서 일부 코드가 주석 처리되지 않아 발생한 문법 오류 수정
- 사용되지 않는 severity 변수 주석 처리로 경고 해결
- 클래스 구조 및 메서드 중괄호 정리
- 모든 컴파일 오류 및 린터 경고 해결

* refactor: 2차 검증 수정 (#142)

* refactor: 유사도 검색 대상 변경

기존 approved, public 에서
denied, blacklist로 변경

* refactor: 블랙리스트, 반려 템플릿 검색 기능 추가

* feat: violations, evidence 중복 제거

* refactor: 검증 최종 수정 (#143)

* fix: merge할때 생겼던 수정사항 커밋'

* fix: merge 충돌 해결

* fix: merge 충돌 해결

* fix: 템플릿 저장 403 오류 해결 - JSON 파싱 문제 수정

- 문제: 템플릿 저장 시 403 Forbidden 오류 발생
- 원인: JWT 인증은 정상이지만 JSON 파싱 실패
  - Jackson이 variableList를 ArrayList<Object>로 파싱하려 하지만 Object 타입을 받음
  - TemplateSaveRequestDto에 @Setter 누락으로 데이터 바인딩 실패

수정 사항:
- TemplateSaveRequestDto에 @Setter 어노테이션 추가
- @JsonProperty 어노테이션으로 JSON 필드 매핑 명시
- setVariableList 메서드로 유연한 데이터 타입 처리
  - List, String[], Object 등 다양한 타입을 List<String>으로 변환
- JWT 인증 필터에 상세한 디버깅 로그 추가
- 프론트엔드 API 인터셉터에 토큰 상태 로깅 강화
- 사용자 스토어의 토큰 복원 로직 개선

테스트:
- JWT 인증 성공 확인
- JSON 파싱 오류 해결
- 템플릿 저장 기능 정상화

* fix: 반려 사이드바 높이 고정

* feat: validator 설정 및 유틸리티 파일 추가

* refactor: 사용하지 않는 프롬프트 파일 제거 및 __init__.py 업데이트

* refactor: validator 핵심 로직 개선 및 파이프라인 최적화

* feat: ChromaDB 서비스 기능 개선

* fix: 상대 import 오류 해결 - 절대 import로 변경

- alimtalk_routes.py: 상대 import를 절대 import로 수정
- alimtalk_service.py: 상대 import를 절대 import로 수정
- constraint_validator.py: 상대 import를 절대 import로 수정
- validator_pipeline.py: 상대 import를 절대 import로 수정

* feat: AI 검증 시스템 모델 및 서비스 개선

- ProblemArea 모델 추가: 문제 영역 식별 및 추적을 위한 상세 모델 구현
- ValidationResponse 구조 개선: 문제 영역 기반 응답 형식으로 변경
- AlimtalkValidationService 배치 처리 최적화: 다중 오류에 대한 효율적인 대안 생성
- AI 기반 문제 영역 추출 및 대안 생성 로직 강화
- 검증 결과의 상세 정보 제공 및 사용자 친화적 메시지 개선

* feat: 검증 프롬프트 시스템 개선

- 시스템 프롬프트 정확도 향상: 모호한 부분 허용 방향으로 판단 기준 개선
- 템플릿 작성 규칙 검증 로직 개선: 연령인증 관련 표현 검증 조건 명확화
- 변수 사용 규칙 검증 강화: 템플릿 본문과 버튼/미리보기 구분 명확화
- 허용되는 표현과 금지되는 표현 기준 세분화
- JSON 응답 형식 표준화 및 일관성 개선

* feat: 의미적 검증기 개선

- SemanticValidator RAG 기반 검증 로직 강화: 반려 사유 DB 검색 정확도 향상
- 블랙리스트 룰 기반 검증 추가: 정규식 패턴 매칭을 통한 금지 표현 검출
- 위험도 계산 알고리즘 개선: 최고 유사도와 평균 유사도 가중평균 적용
- 중복 제거 로직 구현: evidence와 source 조합으로 중복 판별
- 검증 결과 상세 정보 제공: 위반 사항, 근거, 분석 결과 포함

* feat: 프론트엔드 UI/UX 개선

- KakaoPreviewComponent 템플릿 렌더링 최적화: 마커 제거 및 변수 표시 개선
- RejectionSidebarComponent 대안 선택 UI 개선: 문제 영역별 대안 표시 및 선택 로직 강화
- TemplateResultView 문제 영역 편집 시스템 구현: ID 마커 기반 다중 수정 지원
- 문제 영역 하이라이트 및 수정 추적 기능 추가
- 사용자 친화적 오류 메시지 및 안내 개선

* feat: AI 프롬프트 시스템 추가

- 문제 영역 추출 프롬프트 추가: AI 기반 템플릿 문제 영역 자동 식별
- 배치 대안 생성 프롬프트 추가: 다중 오류에 대한 효율적인 대안 생성
- 검증 정확도 향상을 위한 프롬프트 최적화
- JSON 응답 형식 표준화 및 파싱 안정성 개선
- 템플릿 검증 워크플로우 자동화 강화

* fix: TemplateResultView.vue 오류 수정

- savedTemplateId 변수 누락 오류 수정: 저장된 템플릿 ID 추적을 위한 ref 변수 추가
- findStablePosition 함수 누락 오류 수정: 다중 수정을 위한 안정적인 위치 찾기 함수 구현
- 문제 영역 편집 시스템의 안정성 향상: 문맥 기반 위치 찾기 및 백업 위치 찾기 로직 추가
- 템플릿 ID 관리 기능 개선: 저장된 템플릿 참조를 위한 상태 관리 강화

* feat: AI 프롬프트 시스템 개선

- 템플릿 생성 프롬프트의 변수 처리 규칙 강화
- 대안 생성 프롬프트의 배치 처리 기능 추가
- 문제 영역 추출 프롬프트의 정확도 향상
- 변수명 표준화 및 일관성 개선

* feat: 템플릿 미리보기 및 결과 화면 UI 개선

- 카카오톡 미리보기 컴포넌트의 텍스트 정리 로직 개선
- 문제 영역 하이라이트 기능 강화
- 템플릿 수정 및 검증 플로우 최적화
- 사용자 경험 향상을 위한 인터랙션 개선

* fix: 과도한 로깅 정리

* refactor: TemplateService 구문 오류 수정 및 JwtAuthenticationFilter 중첩 if문 리팩토링

- TemplateService.java: if문 중괄호 누락으로 인한 구문 오류 수정
- JwtAuthenticationFilter.java: Early Return 패턴 적용으로 중첩 if문 제거
  - 각 인증 단계를 순차적으로 처리하여 가독성 향상
  - 로깅 로직을 별도 메서드로 분리
  - 코드 유지보수성 및 디버깅 용이성 개선

---------

Co-authored-by: SeungminShin97 <107799398+SeungminShin97@users.noreply.github.com>

Commit:
chore: 타입 에러 해결 및 병렬처리(설계 수정)

Commit:
chore: 프롦프트 템플릿 수정

Commit:
chore: 변수 덩어리로 키, 벨류로 매핑

choichanwoo001 committed:
Merge branch 'template_pt' of github.com:Kernel180-BE12/final_6team_pls_jober into template_pt

choichanwoo001 committed:
Merge branch 'template-pt-hoyoung-new' of github.com:Kernel180-BE12/final_6team_pls_jober into template-pt-hoyoung-new

choichanwoo001 committed:
fix: SuitabilityCheckPromptBuilder 클래스 누락 문제 해결

- builders.py에 SuitabilityCheckPromptBuilder 클래스 추가
- 클래스 정의 순서 수정으로 NameError 해결
- BasePromptBuilder에 _apply_security_protection 메서드 추가
- __init__.py에서 SuitabilityCheckPromptBuilder export 추가
- templateEngine 파이프라인 시작 시 ImportError 해결

choichanwoo001 committed:
fix: 미리보기 템플릿 내용 깨짐 현상 해결

- KakaoPreviewComponent에서 과도한 텍스트 정리 로직 제거
- 40개 이상의 정규식으로 인한 텍스트 손실 방지
- 원본 템플릿 내용 보존을 위한 최소한의 정리만 수행
- 마커 제거와 빈 줄 정리만 유지하여 내용 일치성 확보
- 디버깅 로그 추가로 템플릿 처리 과정 추적 가능

choichanwoo001 committed:
fix: AI 서비스 merge conflict 해결 및 중복 파일 정리

- ai/services/chromadb_service.py: 컬렉션 관리 구조 통합
- ai/api/routes/template_routes.py: Session import 추가 및 파라미터 수정
- ai/templateEngine/nodes.py: DB 연동 병렬 처리 노드 통합
- ai/templateEngine/pipeline.py: 워크플로우 흐름 재구성
- ai/templateEngine/prompts/builders.py: 프롬프트 빌더 구조 통합
- ai/validators/semantic_validator.py: RAG 검색 로직 통합
- 중복 파일 삭제: nodes_optimized.py, pipeline_optimized.py, builders_optimized.py

choichanwoo001 committed:
refactor: 중복 클래스 제거 및 프론트엔드 오류 수정

- 템플릿 엔진 중복 클래스 정리
  * builders.py와 message_analyzer_prompts.py의 중복된 PromptBuilder 클래스들 제거
  * BasePromptBuilder, TypePromptBuilder, FieldsPromptBuilder 등 통합
  * message_analyzer_prompts.py에서 builders.py import로 변경

- 프론트엔드 CSS 구문 오류 수정
  * TemplateResultView.vue에서 JavaScript 스타일 주석(//) 사용 오류 수정
  * CSS 주석 형식(/* */)으로 변경하여 linter 오류 해결

- 템플릿 엔진 코드 정리
  * pipeline.py에서 불필요한 import 및 주석 처리된 코드 제거
  * 코드 구조 개선 및 가독성 향상

- 모든 linter 오류 해결 및 코드 품질 개선

choichanwoo001 committed:
fix: import 오류 및 Pydantic 검증 오류 수정

- pipeline.py Session 타입 import 추가
  * sqlalchemy.orm에서 Session 타입 import 누락으로 인한 NameError 해결

- semantic_validator.py Git merge conflict 마커 제거
  * <<<<<<< HEAD 마커가 남아있어 발생한 IndentationError 해결

- template_routes.py Pydantic 검증 오류 수정
  * category 필드가 None 값을 받을 때 발생하는 ValidationError 해결
  * result.get(category_sub) or 기타로 None 값 처리 개선

Commit:
chore: 파이프라인 수정 및 변수 처리(벨류 값을 템플릿 화면에 보이도록)
구현

choichanwoo001 committed:
style: 새로운 레이아웃에 맞춰 카카오 미리보기 컴포넌트 스타일 업데이트

- 1:2 레이아웃 비율에 맞게 컴포넌트 스타일 조정
- 업데이트된 오른쪽 패널 디자인과 시각적 일관성 확보
- 통합된 미리보기 영역 내에서 적절한 통합 유지

choichanwoo001 committed:
feat: 채팅/미리보기 레이아웃 및 UI 스타일 개선

- 레이아웃 비율을 1:1에서 1:2(채팅:미리보기)로 변경하여 공간 활용도 향상
- 채팅 이력과 입력란을 단일 컨테이너로 통합하고 배경색으로 구분
- 패널 간 구분선 위치와 높이 조정으로 정확한 정렬 구현
- 구분선 주변 적절한 간격을 위한 패널 패딩 추가
- 액션 버튼을 미리보기 영역 내부로 재배치
- 채팅 스크롤바 스타일링 및 메시지와의 간격 조정
- 일관된 파란색 테마 유지 (#1976d2)

choichanwoo001 committed:
fix: 템플릿 제목 길이 제한 오류 수정 및 버튼 스타일 통일

- 템플릿 제목 최대 길이를 50자에서 200자로 확장하여 긴 제목 처리 가능
- 제출하기 버튼의 border-radius를 1rem으로 변경하여 정정 횟수 표시와 스타일 통일
- UI 일관성 향상 및 사용자 경험 개선

choichanwoo001 committed:
Merge branch 'template-pt-minyeong' of github.com:Kernel180-BE12/final_6team_pls_jober into template-pt-hoyoung-new

choichanwoo001 committed:
fix: 누락된 프롬프트 빌더 클래스 추가 및 미리보기 UI 개선

- SuitabilityCheckPromptBuilder, ExpertTemplateBuilder, TemplateWriterBuilder 클래스 추가
- ImportError 해결을 위한 __init__.py 업데이트
- KakaoPreviewComponent에서 접기/자세히 보기 기능 제거
- 미리보기 영역 너비를 16글자 너비(320px)로 조정

choichanwoo001 committed:
feat: 템플릿 생성 파이프라인 및 검증 시스템 개선

- 적합성 검사 로직 추가로 부적절한 메시지 사전 필터링
- 템플릿 생성 파이프라인 구조 개선 및 변수 처리 로직 수정
- 프롬프트 빌더 클래스들 개선 및 새로운 빌더 추가
- 의미적 검증 시스템 로직 개선
- 카카오톡 미리보기 컴포넌트 UI/UX 개선

choichanwoo001 committed:
fix: 저장하지 않은 사항들 저장

choichanwoo001 committed:
fix: vite서버가 경로 인식 못하는 오류 해결

choichanwoo001 committed:
fix: TemplateRequestDto JSON 파싱 오류 해결

- chatHistory와 variableList 필드의 JSON 파싱 오류 수정
- 커스텀 JsonDeserializer 추가로 안전한 JSON 파싱 구현
- @JsonIgnoreProperties(ignoreUnknown = true) 추가로 알 수 없는 필드 무시
- List 필드들을 기본값으로 초기화하여 NullPointerException 방지

choichanwoo001 committed:
refactor: 모든 variableList를 딕셔너리 배열 구조로 통일

- 백엔드 DTO들 (TemplateRequestDto, TemplateValidationRequestDto, TemplateSaveRequestDto)에서 variableList를 List<Map<String, String>>로 변경
- 커스텀 JsonDeserializer 추가로 문자열 배열과 딕셔너리 배열 모두 처리 가능
- 프론트엔드에서 convertToStringArray 헬퍼 함수로 딕셔너리 배열 변환
- AI Python 모델에서 TemplateModificationRequest의 variableList를 List[Dict[str, str]]로 수정
- TemplateService에서 딕셔너리 구조 처리 로직 수정

choichanwoo001 committed:
fix: variable - key 매핑 dict 형태로 수정

choichanwoo001 committed:
refactor: 공백 추가
